src/sql/workbench/browser/modelComponents/treeComponentRenderer.ts

```diff
@@ -160,8 +167,13 @@ export class TreeComponentRenderer extends Disposable implements IRenderer {
 		let label = treeNode.label;
 		templateData.label.textContent = label.label;
 		templateData.root.title = label.label;
-		templateData.checkboxState = this.getCheckboxState(treeNode);
-		templateData.enableCheckbox = treeNode.enabled;
+
+		if (templateData.checkbox) {
```

Comment:
I'm curious how does this work? In the `TreeDataTemplate` above it defines `_checkbox` as `private _checkbox: HTMLInputElement;` but there is nothing assigning a value to it (no constructor, no nothing). So it is `undefined`, even thought type doesn't claim it to be so. Previously code was unconditionally setting checkbox properties, but it may not be set to begin with. Wouldn't that result in "Cannot access... of undefined"?

Technically, this line should trigger tsc, as the condition is always true, but it doesn't. aren't you running in strict mode?

Reply:
https://github.com/microsoft/azuredatastudio/blob/cb7b7da0a4bb2b45f0b6092e42fd1ca385938f09/src/sql/workbench/browser/modelComponents/treeComponentRenderer.ts#L33

This is where _checkbox is set. 

We don't compile in strict mode - by the time that some people started looking at it there was so much debt that enabling that for the core code is pretty much out of the question at this point. (there would be multiple major refactorings that would be needed of extremely central services such as connection management)

Reply:
Yeah, I've seen where it's set, but that has to be set explicitly. There is no guarantee that it won't be undefined here, so this check should be enforced by tsc, but you have disabled TS DFA, at which point you could probably just use the raw JavaScript to begin with... 🤷